src/background/adblocker.js

```diff
@@ -237,6 +312,8 @@ async function injectCosmetics(details, config) {
   const domain = parsed.domain || '';
   const hostname = parsed.hostname || '';
 
+  if (scriptletsOnly && contentScripts.isRegistered(hostname)) return;
```

Comment:
This `scripletsOnly` is not what the name suggests - it is a flag for `EXPERIMENTAL_SCRIPTLETS_INJECTION` - you can't use `scripletsOnly` in Chrome if you would wish for example.

There is an unrelated bug in Firefox requiring the separation of injecting styles and scriptlets (styles must be injected not in `onCommited`, as then we have permission for an issue in iframes). So as this function becomes harder to reason about, this is a moment to do so...

Reply:
not quite right, this block is just an optimization that prevent a matching if content scripts are already registered. `scriptletsOnly` main role is to prevent style injection a bit below.

But yes, the condition should be improved